enhancements/sig-architecture/105-aws-iam-registration/README.md

```diff
@@ -0,0 +1,464 @@
+
+If the feature gate `ManagedClusterAutoApproval` feature gate is enabled, choosing `aws-irsa` instead of `csr` will disable [csr controller](https://github.com/open-cluster-management-io/ocm/blob/main/pkg/registration/hub/csr/controller.go) on hub and enable a new `aws-irsa` controller, which will auto approve all registration requests, and create IAM roles and policies and RBAC permissions. 
+
+The hub registration controller must assume an IAM role with the permissions allowing it to create roles for managed clusters and attach policies. The name of this role will follow a standard template which will contain cluster name which can be parsed from the kubeconfig used to run `clusteradm init` command. The role name should follow a standard name so that the role can be created before running `clusteradm init` command and the hub controller can assume the role with a standardized name, on startup:
+```shell
+arn:aws:iam::<hub-account-id>:role/<hub-cluster-name>_managed-cluster-identity-creator
+```
+
+#### Changes required on the managed cluster
+Choosing `aws-irsa` instead of `csr` will disable [cert_controller](https://github.com/open-cluster-management-io/ocm/blob/main/pkg/registration/clientcert/cert_controller.go) on managed cluster and enable a new `aws-irsa` controller in klusterlet/registration agent, which will create only `ManagedCluster` on hub and not CSR. The `aws-irsa` controller will watch for the updates on `ManagedCluster` , and once accepted on hub and has HubAcceptedManagedCluster condition, it will generate a kubeconfig using following command and save it to a secret called hub-kubeconfig-secret on managed cluster:
+```shell
+aws eks update-kubeconfig --name <hub-eks-cluster-name> --kubeconfig <path/to/create/kubeconfig> --role-arn <role-on-hub-for-managed>
+```
+
+The klusterlet/registration agent on managed cluster must assume an IAM role, in the AWS account in which managed cluster is running, on startup, with the permissions allowing it assume the role on hub IAM. The role will be tagged with managed cluster name which can be parsed from the `--cluster-name` option passed to `clusteradm join` command. The managed cluster admin must know in advance, which hub cluster to join, as the role will be tagged with the name and aws account of the hub cluster.
+
+The role name and policies created on managed cluster IAM will follow a templates to ensure standardization. The role name will contain md5 hash of the values of tags to uniquely identify.
+
+```shell
+arn:aws:iam::<managed-cluster-account-id>:role/ocm-spoke-<md5-hash-unique-identifier>
+```
+
+Another reason to follow a standard name is to ensure that the role can be created before running `clusteradm join` command and the klusterlet/registration agent can assume that role, on startup, with a standardized name.
+
+#### Graceful resource cleanup on both hub and managed cluster when managed cluster triggers unjoin
```

Comment:
Currently the `clusteradm unjoin` command only deletes resources on the managed cluster side, the ManagedCluster CR on the hub side will not be deleted.
I am curious if the IAM roles and policies only need to be deleted when deleting the ManagedCluster, if it is, maybe a finalizer is simpler

Reply:
We want managedcluster specific resources on the hub to be cleaned up even when managed cluster triggers unjoin and managedcluster doesn't have delete permission on its managedcluster CR on hub.